cv_next/server/api/cvs.ts

```diff
@@ -1,277 +1,165 @@
+      return null
+    }
+
+    return cvs as CvModel[]
   } catch (error) {
-    MyLogger.logInfo("Error @ FirebaseHelper::_getAllCvsByQueryFilter", error)
+    MyLogger.logInfo("Error @ getCvsByUserId", error)
+    return null
   }
-  return null
-}
-
-/**
- * Retrieves all CVs by user ID.
- *
- * @param {string} userId - The ID of the user.
- * @param {boolean} filterOutDeleted - Whether to filter out deleted CVs.
- * @return {Promise<CvModel[] | null>} - A promise that resolves to an array of CV models or null.
- */
-export async function getAllCvsByUserId(
-  userId: string,
-  filterOutDeleted: boolean = true
-): Promise<CvModel[] | null> {
-  return _getAllCvsByQueryFilter(
-    where("userID", "==", userId),
-    filterOutDeleted
-  )
 }
 
-/**
- * Retrieves all CVs by category.
- *
- * @param {Categories.category} category - The category to filter CVs by.
- * @param {boolean} filterOutDeleted - Whether to exclude deleted CVs from the result. Defaults to true.
- * @return {Promise<CvModel[] | null>} - A promise that resolves to an array of CV models or null.
- */
 export async function getAllCvsByCategory(
   category: Categories.category,
   filterOutDeleted: boolean = true
 ): Promise<CvModel[] | null> {
-  return _getAllCvsByQueryFilter(
-    where("categoryID", "==", category),
-    filterOutDeleted
-  )
+  try {
+    const supabase = SupabaseHelper.getSupabaseInstance()
+    let query = supabase.from("cvs").select("*").eq("category_id", category)
+
+    if (filterOutDeleted) {
+      query = query.eq("deleted", false)
+    }
+
+    const { data: cvs, error } = await query
+
+    if (error) {
+      MyLogger.logInfo("Error @ getAllCvsByCategory", error)
+      return null
+    }
+
+    return cvs as CvModel[]
+  } catch (error) {
+    MyLogger.logInfo("Error @ getAllCvsByCategory", error)
+    return null
+  }
 }
 
 /**
- * Retrieves paginated CVs from the database.
+ * Retrieves a paginated list of CVs based on the provided page number.
  *
- * @param {boolean} filterOutDeleted - Whether to filter out deleted CVs.
- * @param {string?} lastId - optionally pass the last cv id to start the query after. 
- * @return {Promise<CvModel[] | null>} - A promise that resolves to an array of CV models or null.
+ * @param {boolean} filterOutDeleted - Indicates whether deleted CVs should be filtered out.
+ * @param {number} page - The page number for pagination.
+ * @return {Promise<CvModel[] | null>} A Promise that resolves with an array of CvModel or null.
  */
 export async function getPaginatedCvs(
   filterOutDeleted: boolean = true,
-  lastId?: string
-): Promise<CvModel[] | null> {
-  let res = getPaginatedCvsByQueryFilter(undefined, filterOutDeleted, lastId)
-  MyLogger.logDebug(
-    `requested all cvs(pagination) - last cv id: ${lastId}`
-  )
-  return res
+  page?: number
+): Promise<PaginatedCvsModel | null> {
+  try {
+    page = page ?? Definitions.DEFAULT_PAGINATION_FIRST_PAGE_NUMBER
```

Comment:
Instead of doing this you can default-assign page to the default (i.e. in function definition). Also accordingly you can call the function without providing `undefined` but simply pass nothing.